src/components/TracePage/TraceTimelineViewer/SpanDetailRow.js

```diff
@@ -37,25 +65,25 @@ export default function SpanDetailRow(props) {
           <span>
             <span
               className="detail-row-expanded-accent"
-              onClick={toggleDetailExpansion}
+              onClick={detailToggle}
               style={{ borderColor: color }}
             />
           </span>
         </div>
       </TimelineRow.Left>
       <TimelineRow.Right>
         <div className="p2 detail-info-wrapper" style={{ borderTopColor: color }}>
-          <SpanDetail span={span} trace={trace} />
+          <SpanDetail
```

Comment:
I think a higher level API that could be composed to these props is.
```js
<SpanDetail trace={$Trace} spanID={$spanID} />
```
which can call this component you have

That way its a similar data model

Reply:
Only thing needed from the trace was the start time, so I changed the param to `traceStartTime` and kept the rest as-is. Most of them (`*Toggle`) trigger actions.